outbox-kafka-spring/src/main/java/one/tomorrow/transactionaloutbox/repository/OutboxLockRepository.java

```diff
@@ -15,78 +15,90 @@
-import jakarta.persistence.PersistenceContext;
-import jakarta.persistence.TypedQuery;
 import lombok.AllArgsConstructor;
-import lombok.NoArgsConstructor;
 import one.tomorrow.transactionaloutbox.model.OutboxLock;
-import org.hibernate.LockOptions;
-import org.hibernate.dialect.lock.LockingStrategyException;
-import org.hibernate.exception.ConstraintViolationException;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.dao.DuplicateKeyException;
+import org.springframework.jdbc.UncategorizedSQLException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.ResultSetExtractor;
 import org.springframework.stereotype.Repository;
+import org.springframework.transaction.PlatformTransactionManager;
+import org.springframework.transaction.TransactionStatus;
 import org.springframework.transaction.annotation.Propagation;
 import org.springframework.transaction.annotation.Transactional;
 import org.springframework.transaction.interceptor.TransactionAspectSupport;
+import org.springframework.transaction.support.TransactionTemplate;
+import org.springframework.util.StringUtils;
 
+import java.sql.Timestamp;
 import java.time.Duration;
 import java.time.Instant;
-import java.util.Map;
 import java.util.Optional;
 
 import static java.time.Instant.now;
+import static one.tomorrow.transactionaloutbox.model.OutboxLock.OUTBOX_LOCK_ID;
 
 @Repository
-@NoArgsConstructor
-@AllArgsConstructor
 public class OutboxLockRepository {
 
     private static final Logger logger = LoggerFactory.getLogger(OutboxLockRepository.class);
+    private static final ResultSetExtractor<OutboxLock> resultSetExtractor = rs ->
+            rs.next()
+                    ? new OutboxLock(rs.getString("owner_id"), rs.getTimestamp("valid_until").toInstant())
+                    : null;
 
-    @PersistenceContext
-    private EntityManager entityManager;
+    private final JdbcTemplate jdbcTemplate;
+    private final TransactionTemplate transactionTemplate;
+
+    public OutboxLockRepository(JdbcTemplate jdbcTemplate, PlatformTransactionManager transactionManager) {
+        this.jdbcTemplate = jdbcTemplate;
+        this.transactionTemplate = new TransactionTemplate(transactionManager);
+    }
 
     @Transactional(propagation = Propagation.REQUIRES_NEW)
     public boolean acquireOrRefreshLock(String ownerId, Duration timeout) {
+        return transactionTemplate.execute(status -> acquireOrRefreshLock(ownerId, timeout,  status));
+    }
+
+    private boolean acquireOrRefreshLock(String ownerId, Duration timeout, TransactionStatus txStatus) {
 
-        OutboxLock lock = null;
         Instant now = now();
+        Instant validUntil;
         try {
-            lock = entityManager.find(OutboxLock.class, OutboxLock.OUTBOX_LOCK_ID);
+            OutboxLock lock = jdbcTemplate.query("select * from outbox_kafka_lock where id = '" + OUTBOX_LOCK_ID + "'", resultSetExtractor);
```

Comment:
Right now the compiler will create the String, while with String.format this would happen at runtime, without any advantage. That's why I prefer the static / compile time way.

Reply:
@hnrkdmsk makes sense?